### ISSUE TITLE
How to find the optimal parameters quickly，grid search?

### ISSUE THREAD

**Issue by gongshaojie12** [facebook/prophet#671]
I train a prophet model，How can I quickly find the optimal parameters of this model?

**Comment by markrazmandi** [comment]
Having an understanding of the data and the effect/interdependence of parameters within the model will narrow the scope of parameter combinations. A grid search can then be implemented over a cluster for fastest processing. Depending on your problem, you should understand the error metrics that best represent the overall accuracy, stability, and variance in the model. Brute force tactics can work to the extent you have the computation power to handle.

**Comment by bletham** [comment]
The prior scales operate pretty independently, so I agree with @markrazmandi that in the ideal case you would be able to do this in-the-loop and figure out what is best for your dataset. When you have too many datasets for that to be reasonable than a hyperparameter sweep could be reasonable, but allow me to take a minute to say that grid search is quite inefficient and random search is generally better (and sometimes much better). Fig. 1 of this paper says pretty much all you need to know about grid search: http://www.jmlr.org/papers/volume13/bergstra12a/bergstra12a.pdf

**Comment by leeprevost** [comment]
I have this need too.  A work around for this is to use sklearn ParameterGrid.

```
from sklearn.model_selection import ParameterGrid
params_grid = {'growth':('linear', 'logistic'),
               'changepoint_prior_scale':[0.3, 0.7]}
grid = ParameterGrid(params_grid)
print([p for p in grid])
```

> [{'changepoint_prior_scale': 0.3, 'growth': 'linear'}, {'changepoint_prior_scale': 0.3, 'growth': 'logistic'}, {'changepoint_prior_scale': 0.7, 'growth': 'linear'}, {'changepoint_prior_scale': 0.7, 'growth': 'logistic'}]

then iterate through the grid iterator:
```
for p in grid:
    m =Prophet(**p)
    m.fit(data)
    #do scoring or diag
    #save p and score
```
